### ISSUE TITLE
Make the tests run

### ISSUE THREAD

**Issue by xR3b0rn** [zherczeg/sljit#45]
I do have real problems making the tests run.
I saw the last changes to the tests were made 5 years ago. Is there somewhere an updated version of a documentation?

Using the command:
```
gcc -Wall -g -DSLJIT_CONFIG_AUTO=1 -I../../sljit_src/ ../../sljit_src/sljitLir.c func_call.c
```
Gives the error:
```
In file included from ../../sljit_src/sljitLir.h:81,
                 from ../../sljit_src/sljitLir.c:27:
../../sljit_src/sljitConfigInternal.h:219:2: error: #error "-mshstk is needed to compile with -fcf-protection"
  219 | #error "-mshstk is needed to compile with -fcf-protection"
      |  ^~~~~
In file included from ../../sljit_src/sljitLir.h:81,
                 from func_call.c:1:
```
Adding `-mshstk` leads to this output:
```
In file included from func_call.c:1:
../../sljit_src/sljitLir.h:1211:22: error: expected identifier or ‘(’ before numeric constant
 1211 | #define SLJIT_CALL   26
      |                      ^~
func_call.c:6:14: note: in expansion of macro ‘SLJIT_CALL’
    6 | typedef long SLJIT_CALL (*func3_t)(long a, long b, long c);
      |              ^~~~~~~~~~
```
In which the affected area in the source code looks realy strange to me:
```C++
/* Unconditional jump types. */
#define SLJIT_JUMP			24
	/* Fast calling method. See sljit_emit_fast_enter / SLJIT_FAST_RETURN. */
#define SLJIT_FAST_CALL			25
	/* Called function must be declared with the SLJIT_FUNC attribute. */
#define SLJIT_CALL			26
	/* Called function must be declared with cdecl attribute.
	   This is the default attribute for C functions. */
#define SLJIT_CALL_CDECL		27
```
Removing the macros from the function declarations makes the example compile. But then some assertions will trigger, when removing them, calling the function ends in seg fault.

Are there are somewhere some up to date tutorials?
Or am I doing just a small little thing wrong?

Thanks.

**Comment by xR3b0rn** [comment]
In I've found `SLJIT_CALL` there: https://github.com/zherczeg/sljit/blob/ddbe95d600be0d9fa492f60f0b369afc27166fc6/sljit_src/sljitConfigInternal.h
in a very old commit.

**Comment by zherczeg** [comment]
You are right, those examples are outdated, although they should work with some minor changes. SLJIT_CALL is renamed to SLJIT_FUNC. Furthermore GCC in the latest Ubuntu uses -fcf-protection by default but it does not pass -mshstk, and you need to pass it manually or disable -fcf-protection. These two options enables Intel Control Flow Enforcement technique. If you could make those tests work, could you create a PR with the changes?

**Comment by zherczeg** [comment]
Fixed in #86
